skbio/metadata/_mixin.py

```diff
@@ -406,3 +407,114 @@ def has_positional_metadata(self):
         """
         return (self._positional_metadata is not None and
                 len(self.positional_metadata.columns) > 0)
+
+
+class IntervalMetadataMixin(metaclass=abc.ABCMeta):
+    def _interval_metadata_axis_len_(self):
+        ''''''
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def __init__(self, interval_metadata=None):
+        raise NotImplementedError
+
+    def _init_(self, interval_metadata=None):
+        if interval_metadata is None:
+            # Could use deleter but this is less overhead and needs to be fast.
+            self._interval_metadata = None
+        else:
+            # Use setter for validation and copy.
+            self.interval_metadata = interval_metadata
+
+    @property
+    @experimental(as_of="0.5.0-dev")
+    def interval_metadata(self):
+        '''``IntervalMetadata`` object containing info about interval features.
+        '''
+        if self._interval_metadata is None:
+            # Not using setter to avoid copy.
+            self._interval_metadata = IntervalMetadata(
+                self._interval_metadata_axis_len_())
+        return self._interval_metadata
+
+    @interval_metadata.setter
+    def interval_metadata(self, interval_metadata):
+        if isinstance(interval_metadata, IntervalMetadata):
+            upper_bound = interval_metadata.upper_bound
+            axis_len = self._interval_metadata_axis_len_()
+            if upper_bound != axis_len:
```

Comment:
Can you also check that `.lower_bound` is zero? Currently it will always be zero but that may not be true in the future, and we'd want to catch that here.